### ISSUE TITLE
`DisableUrlCacheTest.testReloadWebAppWithLog4j2()` is flaky 

### ISSUE THREAD

**Issue by lorban** [eclipse/jetty.project#11662]
**Jetty version(s)**
12.0.x

**Jetty Environment**
ee10

**Description**
`org.eclipse.jetty.ee10.tests.distribution.DisableUrlCacheTest.testReloadWebAppWithLog4j2()` sometimes fails with:

```
java.lang.AssertionError: 
Expected: is <200>
     but: was <404>
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:6)
	at org.eclipse.jetty.ee10.tests.distribution.DisableUrlCacheTest.testReloadWebAppWithLog4j2(DisableUrlCacheTest.java:108)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at java.base/java.util.concurrent.RecursiveAction.exec(RecursiveAction.java:194)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:387)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1312)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1843)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1808)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:188)
```

See: https://jenkins.webtide.net/blue/organizations/jenkins/jetty.project/detail/PR-11637/7/tests

**Comment by joakime** [comment]
Weird that it's a 404, that seems to means the webapp is there, but didn't match any url-pattern on the request path??

If the webapp didn't redeploy properly (a possibility) I would have expected to see a 503 (unavailable) instead.

**Comment by joakime** [comment]
Adding logging output when this occurred (to preserve it, as once the PR is merged, this information gets deleted on CI)

```
Running org.eclipse.jetty.ee10.tests.distribution.DisableUrlCacheTest.testReloadWebAppWithLog4j2()
2024-04-17 16:09:35.890:INFO :org.eclipse.jetty.tests.testers.JettyHomeTester:ForkJoinPool-1-worker-2: Executing: [/home/jenkins/agent/tools/hudson.model.JDK/jdk21/bin/java, -Djava.io.tmpdir=/home/jenkins/agent/workspace/jetty.project_PR-11637/tests/test-distribution/test-ee10-distribution/target/tests/oejetd.DisableUrlCacheTest/testReloadWebAppWithLog4j2/work, -jar, /home/jenkins/agent/workspace/jetty.project_PR-11637/tests/test-distribution/test-ee10-distribution/target/tests/test-homes/1713364164000/jetty-home-12.0.9-SNAPSHOT/start.jar, --add-to-start=http,ee10-webapp,ee10-deploy,disable-urlcache, maven.local.repo=/home/jenkins/agent/workspace/jetty.project_PR-11637/.repository, maven.repo.uri=http://nexus-service.nexus.svc.cluster.local:8081/repository/maven-public/]
2024-04-17 16:09:35.890:INFO :org.eclipse.jetty.tests.testers.JettyHomeTester:ForkJoinPool-1-worker-2: Working Dir: /home/jenkins/agent/workspace/jetty.project_PR-11637/tests/test-distribution/test-ee10-distribution/target/tests/oejetd.DisableUrlCacheTest/testReloadWebAppWithLog4j2
2024-04-17 16:09:36.264:INFO :org.eclipse.jetty.tests.testers.ProcessWrapper:process/err: WARN  : Option --add-to-start is deprecated! Instead use: --add-modules=http,ee10-webapp,ee10-deploy,disable-urlcache
2024-04-17 16:09:36.345:INFO :org.eclipse.jetty.tests.testers.ProcessWrapper:process/err: INFO  : mkdir ${jetty.base}/start.d
2024-04-17 16:09:36.347:INFO :org.eclipse.jetty.tests.testers.ProcessWrapper:process/err: INFO  : server          transitively enabled, ini template available with --add-modules=server
2024-04-17 16:09:36.347:INFO :org.eclipse.jetty.tests.testers.ProcessWrapper:process/err: INFO  : sessions        transitively enabled, ini template available with --add-modules=sessions
2024-04-17 16:09:36.348:INFO :org.eclipse.jetty.tests.testers.ProcessWrapper:process/err: INFO  : disable-urlcache initialized in ${jetty.base}/start.d/disable-urlcache.ini
2024-04-17 16:09:36.348:INFO :org.eclipse.jetty.tests.testers.ProcessWrapper:process/err: INFO  : ee10-security   transitively enabled
2024-04-17 16:09:36.349:INFO :org.eclipse.jetty.tests.testers.ProcessWrapper:process/err: INFO  : ee-webapp       transitively enabled, ini template available with --add-modules=ee-webapp
2024-04-17 16:09:36.349:INFO :org.eclipse.jetty.tests.testers.ProcessWrapper:process/err: INFO  : resources       transitively enabled
2024-04-17 16:09:36.349:INFO :org.eclipse.jetty.tests.testers.ProcessWrapper:process/err: INFO  : threadpool      transitively enabled, ini template available with --add-modules=threadpool
2024-04-17 16:09:36.349:INFO :org.eclipse.jetty.tests.testers.ProcessWrapper:process/err: INFO  : deploy          transitively enabled
2024-04-17 16:09:36.349:INFO :org.eclipse.jetty.tests.testers.ProcessWrapper:process/err: INFO  : logging-jetty   transitively enabled
2024-04-17 16:09:36.349:INFO :org.eclipse.jetty.tests.testers.ProcessWrapper:process/err: INFO  : security        transitively enabled
2024-04-17 16:09:36.350:INFO :org.eclipse.jetty.tests.testers.ProcessWrapper:process/err: INFO  : ee10-deploy     initialized in ${jetty.base}/start.d/ee10-deploy.ini
2024-04-17 16:09:36.351:INFO :org.eclipse.jetty.tests.testers.ProcessWrapper:process/err: INFO  : ee10-webapp     initialized in ${jetty.base}/start.d/ee10-webapp.ini
2024-04-17 16:09:36.351:INFO :org.eclipse.jetty.tests.testers.ProcessWrapper:process/err: INFO  : http            initialized in ${jetty.base}/start.d/http.ini
2024-04-17 16:09:36.352:INFO :org.eclipse.jetty.tests.testers.ProcessWrapper:process/err: INFO  : ee10-servlet    transitively enabled
2024-04-17 16:09:36.352:INFO :org.eclipse.jetty.tests.testers.ProcessWrapper:process/err: INFO  : logging/slf4j   transitive provider of logging/slf4j for logging-jetty
2024-04-17 16:09:36.352:INFO :org.eclipse.jetty.tests.testers.ProcessWrapper:process/err: INFO  : logging/slf4j   dynamic dependency of logging-jetty
2024-04-17 16:09:36.352:INFO :org.eclipse.jetty.tests.testers.ProcessWrapper:process/err: INFO  : bytebufferpool  transitively enabled, ini template available with --add-modules=bytebufferpool
2024-04-17 16:09:36.352:INFO :org.eclipse.jetty.tests.testers.ProcessWrapper:process/err: INFO  : mkdir ${jetty.base}/resources
2024-04-17 16:09:36.352:INFO :org.eclipse.jetty.tests.testers.ProcessWrapper:process/err: INFO  : mkdir ${jetty.base}/webapps
2024-04-17 16:09:36.353:INFO :org.eclipse.jetty.tests.testers.ProcessWrapper:process/err: INFO  : copy ${jetty.home}/modules/logging/jetty/resources/jetty-logging.properties to ${jetty.base}/resources/jetty-logging.properties
2024-04-17 16:09:36.354:INFO :org.eclipse.jetty.tests.testers.ProcessWrapper:process/err: INFO  : Base directory was modified
2024-04-17 16:09:36.392:INFO :org.eclipse.jetty.tests.testers.JettyHomeTester:ForkJoinPool-1-worker-2: Executing: [/home/jenkins/agent/tools/hudson.model.JDK/jdk21/bin/java, -Djava.io.tmpdir=/home/jenkins/agent/workspace/jetty.project_PR-11637/tests/test-distribution/test-ee10-distribution/target/tests/oejetd.DisableUrlCacheTest/testReloadWebAppWithLog4j2/work, -jar, /home/jenkins/agent/workspace/jetty.project_PR-11637/tests/test-distribution/test-ee10-distribution/target/tests/test-homes/1713364164000/jetty-home-12.0.9-SNAPSHOT/start.jar, jetty.http.port=39989, jetty.deploy.scanInterval=1, maven.local.repo=/home/jenkins/agent/workspace/jetty.project_PR-11637/.repository, maven.repo.uri=http://nexus-service.nexus.svc.cluster.local:8081/repository/maven-public/]
2024-04-17 16:09:36.392:INFO :org.eclipse.jetty.tests.testers.JettyHomeTester:ForkJoinPool-1-worker-2: Working Dir: /home/jenkins/agent/workspace/jetty.project_PR-11637/tests/test-distribution/test-ee10-distribution/target/tests/oejetd.DisableUrlCacheTest/testReloadWebAppWithLog4j2
2024-04-17 16:09:37.347:INFO :org.eclipse.jetty.tests.testers.ProcessWrapper:process/err: 2024-04-17 16:09:37.344:WARN :oejx.XmlConfiguration:main: Deprecated method public static void org.eclipse.jetty.ee.WebAppClassLoading.addHiddenClasses(org.eclipse.jetty.util.Attributes,java.lang.String[]) in file:///home/jenkins/agent/workspace/jetty.project_PR-11637/tests/test-distribution/test-ee10-distribution/target/tests/test-homes/1713364164000/jetty-home-12.0.9-SNAPSHOT/etc/jetty-ee-webapp.xml
2024-04-17 16:09:37.396:INFO :org.eclipse.jetty.tests.testers.ProcessWrapper:process/err: 2024-04-17 16:09:37.396:INFO :oejs.Server:main: jetty-12.0.9-SNAPSHOT; built: 2024-04-17T12:21:42.230Z; git: 14b3552db7c6a4143ee07233403f3e61d4081406; jvm 21.0.2+13-LTS
2024-04-17 16:09:37.413:INFO :org.eclipse.jetty.tests.testers.ProcessWrapper:process/err: 2024-04-17 16:09:37.413:INFO :oejdp.ScanningAppProvider:main: Deployment monitor ee10 in [file:///home/jenkins/agent/workspace/jetty.project_PR-11637/tests/test-distribution/test-ee10-distribution/target/tests/oejetd.DisableUrlCacheTest/testReloadWebAppWithLog4j2/webapps/] at intervals 1s
2024-04-17 16:09:37.425:INFO :org.eclipse.jetty.tests.testers.ProcessWrapper:process/err: 2024-04-17 16:09:37.425:INFO :oejd.DeploymentManager:main: addApp: App@355ce81c[ee10,null,/home/jenkins/agent/workspace/jetty.project_PR-11637/tests/test-distribution/test-ee10-distribution/target/tests/oejetd.DisableUrlCacheTest/testReloadWebAppWithLog4j2/webapps/test.xml]
2024-04-17 16:09:37.615:INFO :org.eclipse.jetty.tests.testers.ProcessWrapper:process/err: 2024-04-17 16:09:37.615:INFO :oejew.StandardDescriptorProcessor:main: NO JSP Support for /test, did not find org.eclipse.jetty.ee10.jsp.JettyJspServlet
2024-04-17 16:09:37.993:INFO :org.eclipse.jetty.tests.testers.ProcessWrapper:process/err: 2024-04-17 16:09:37.992:INFO :oejsh.ContextHandler:main: Started oeje10w.WebAppContext@a486d78{test,/test,b=file:///home/jenkins/agent/workspace/jetty.project_PR-11637/tests/test-distribution/test-ee10-distribution/target/tests/oejetd.DisableUrlCacheTest/testReloadWebAppWithLog4j2/work/test/webapp/,a=AVAILABLE,h=oeje10s.SessionHandler@cdc3aae{STARTED}}{/home/jenkins/agent/workspace/jetty.project_PR-11637/tests/test-distribution/test-ee10-distribution/target/tests/oejetd.DisableUrlCacheTest/testReloadWebAppWithLog4j2/webapps/test.war}
2024-04-17 16:09:37.998:INFO :org.eclipse.jetty.tests.testers.ProcessWrapper:process/out: 2024-04-17 16:09:37.996:INFO :org.eclipse.jetty.ee10.log4j2.LogContextListener:main: contextInitialized(): jakarta.servlet.ServletContextEvent[source=ServletContext@oeje10w.WebAppContext@a486d78{test,/test,b=file:///home/jenkins/agent/workspace/jetty.project_PR-11637/tests/test-distribution/test-ee10-distribution/target/tests/oejetd.DisableUrlCacheTest/testReloadWebAppWithLog4j2/work/test/webapp/,a=AVAILABLE,h=oeje10s.SessionHandler@cdc3aae{STARTED}}{/home/jenkins/agent/workspace/jetty.project_PR-11637/tests/test-distribution/test-ee10-distribution/target/tests/oejetd.DisableUrlCacheTest/testReloadWebAppWithLog4j2/webapps/test.war}]
2024-04-17 16:09:38.015:INFO :org.eclipse.jetty.tests.testers.ProcessWrapper:process/out: 2024-04-17 16:09:38.014:INFO :org.eclipse.jetty.ee10.log4j2.LogServlet:main: #### init()
2024-04-17 16:09:38.015:INFO :org.eclipse.jetty.tests.testers.ProcessWrapper:process/err: 2024-04-17 16:09:38.015:INFO :oejes.ServletContextHandler:main: Started oeje10w.WebAppContext@a486d78{test,/test,b=file:///home/jenkins/agent/workspace/jetty.project_PR-11637/tests/test-distribution/test-ee10-distribution/target/tests/oejetd.DisableUrlCacheTest/testReloadWebAppWithLog4j2/work/test/webapp/,a=AVAILABLE,h=oeje10s.SessionHandler@cdc3aae{STARTED}}{/home/jenkins/agent/workspace/jetty.project_PR-11637/tests/test-distribution/test-ee10-distribution/target/tests/oejetd.DisableUrlCacheTest/testReloadWebAppWithLog4j2/webapps/test.war}
2024-04-17 16:09:38.020:INFO :org.eclipse.jetty.tests.testers.ProcessWrapper:process/err: 2024-04-17 16:09:38.020:INFO :oejs.DefaultSessionIdManager:main: Session workerName=node0
2024-04-17 16:09:38.029:INFO :org.eclipse.jetty.tests.testers.ProcessWrapper:process/err: 2024-04-17 16:09:38.026:INFO :oejs.AbstractConnector:main: Started ServerConnector@200a26bc{HTTP/1.1, (http/1.1)}{0.0.0.0:39989}
2024-04-17 16:09:38.031:INFO :org.eclipse.jetty.tests.testers.ProcessWrapper:process/err: 2024-04-17 16:09:38.031:INFO :oejs.Server:main: Started oejs.Server@3c3d9b6b{STARTING}[12.0.9-SNAPSHOT,sto=5000] @1633ms
2024-04-17 16:09:38.145:INFO :org.eclipse.jetty.tests.testers.ProcessWrapper:process/out: 2024-04-17 16:09:38.145:INFO :org.eclipse.jetty.ee10.log4j2.LogServlet:qtp632249781-29: doGet() req=org.eclipse.jetty.ee10.servlet.ServletApiRequest@4d9d40bc
2024-04-17 16:09:40.040:INFO :org.eclipse.jetty.tests.testers.ProcessWrapper:process/err: 2024-04-17 16:09:40.039:INFO :oejd.DeploymentManager:Scanner-0-1: removeApp: App@355ce81c[ee10,oeje10w.WebAppContext@a486d78{test,/test,b=file:///home/jenkins/agent/workspace/jetty.project_PR-11637/tests/test-distribution/test-ee10-distribution/target/tests/oejetd.DisableUrlCacheTest/testReloadWebAppWithLog4j2/work/test/webapp/,a=AVAILABLE,h=oeje10s.SessionHandler@cdc3aae{STARTED}}{/home/jenkins/agent/workspace/jetty.project_PR-11637/tests/test-distribution/test-ee10-distribution/target/tests/oejetd.DisableUrlCacheTest/testReloadWebAppWithLog4j2/webapps/test.war},/home/jenkins/agent/workspace/jetty.project_PR-11637/tests/test-distribution/test-ee10-distribution/target/tests/oejetd.DisableUrlCacheTest/testReloadWebAppWithLog4j2/webapps/test.xml]
2024-04-17 16:09:40.044:INFO :org.eclipse.jetty.tests.testers.ProcessWrapper:process/out: 2024-04-17 16:09:40.043:INFO :org.eclipse.jetty.ee10.log4j2.LogContextListener:Scanner-0-1: contextDestroyed(): jakarta.servlet.ServletContextEvent[source=ServletContext@oeje10w.WebAppContext@a486d78{test,/test,b=file:///home/jenkins/agent/workspace/jetty.project_PR-11637/tests/test-distribution/test-ee10-distribution/target/tests/oejetd.DisableUrlCacheTest/testReloadWebAppWithLog4j2/work/test/webapp/,a=AVAILABLE,h=oeje10s.SessionHandler@cdc3aae{STOPPING}}{/home/jenkins/agent/workspace/jetty.project_PR-11637/tests/test-distribution/test-ee10-distribution/target/tests/oejetd.DisableUrlCacheTest/testReloadWebAppWithLog4j2/webapps/test.war}]
2024-04-17 16:09:40.048:INFO :org.eclipse.jetty.tests.testers.ProcessWrapper:process/err: 2024-04-17 16:09:40.048:INFO :oejes.ServletContextHandler:Scanner-0-1: Stopped oeje10w.WebAppContext@a486d78{test,/test,b=null,a=AVAILABLE,h=oeje10s.SessionHandler@cdc3aae{STOPPED}}{/home/jenkins/agent/workspace/jetty.project_PR-11637/tests/test-distribution/test-ee10-distribution/target/tests/oejetd.DisableUrlCacheTest/testReloadWebAppWithLog4j2/webapps/test.war}
2024-04-17 16:09:40.050:INFO :org.eclipse.jetty.tests.testers.ProcessWrapper:process/err: 2024-04-17 16:09:40.050:INFO :oejd.DeploymentManager:Scanner-0-1: addApp: App@1e33b9e7[ee10,null,/home/jenkins/agent/workspace/jetty.project_PR-11637/tests/test-distribution/test-ee10-distribution/target/tests/oejetd.DisableUrlCacheTest/testReloadWebAppWithLog4j2/webapps/test.xml]
2024-04-17 16:09:40.114:INFO :org.eclipse.jetty.tests.testers.ProcessWrapper:process/err: 2024-04-17 16:09:40.114:INFO :oejew.StandardDescriptorProcessor:Scanner-0-1: NO JSP Support for /test, did not find org.eclipse.jetty.ee10.jsp.JettyJspServlet
2024-04-17 16:09:40.323:INFO :org.eclipse.jetty.tests.testers.ProcessWrapper:process/err: 2024-04-17 16:09:40.322:INFO :oejsh.ContextHandler:Scanner-0-1: Started oeje10w.WebAppContext@1fcbb594{test,/test,b=file:///home/jenkins/agent/workspace/jetty.project_PR-11637/tests/test-distribution/test-ee10-distribution/target/tests/oejetd.DisableUrlCacheTest/testReloadWebAppWithLog4j2/work/test/webapp/,a=AVAILABLE,h=oeje10s.SessionHandler@e664844{STARTED}}{/home/jenkins/agent/workspace/jetty.project_PR-11637/tests/test-distribution/test-ee10-distribution/target/tests/oejetd.DisableUrlCacheTest/testReloadWebAppWithLog4j2/webapps/test.war}
2024-04-17 16:09:40.327:INFO :org.eclipse.jetty.tests.testers.ProcessWrapper:process/out: 2024-04-17 16:09:40.325:INFO :org.eclipse.jetty.ee10.log4j2.LogContextListener:Scanner-0-1: contextInitialized(): jakarta.servlet.ServletContextEvent[source=ServletContext@oeje10w.WebAppContext@1fcbb594{test,/test,b=file:///home/jenkins/agent/workspace/jetty.project_PR-11637/tests/test-distribution/test-ee10-distribution/target/tests/oejetd.DisableUrlCacheTest/testReloadWebAppWithLog4j2/work/test/webapp/,a=AVAILABLE,h=oeje10s.SessionHandler@e664844{STARTED}}{/home/jenkins/agent/workspace/jetty.project_PR-11637/tests/test-distribution/test-ee10-distribution/target/tests/oejetd.DisableUrlCacheTest/testReloadWebAppWithLog4j2/webapps/test.war}]
2024-04-17 16:09:40.332:INFO :org.eclipse.jetty.tests.testers.ProcessWrapper:process/out: 2024-04-17 16:09:40.332:INFO :org.eclipse.jetty.ee10.log4j2.LogServlet:Scanner-0-1: #### init()
2024-04-17 16:09:40.333:INFO :org.eclipse.jetty.tests.testers.ProcessWrapper:process/err: 2024-04-17 16:09:40.332:INFO :oejes.ServletContextHandler:Scanner-0-1: Started oeje10w.WebAppContext@1fcbb594{test,/test,b=file:///home/jenkins/agent/workspace/jetty.project_PR-11637/tests/test-distribution/test-ee10-distribution/target/tests/oejetd.DisableUrlCacheTest/testReloadWebAppWithLog4j2/work/test/webapp/,a=AVAILABLE,h=oeje10s.SessionHandler@e664844{STARTED}}{/home/jenkins/agent/workspace/jetty.project_PR-11637/tests/test-distribution/test-ee10-distribution/target/tests/oejetd.DisableUrlCacheTest/testReloadWebAppWithLog4j2/webapps/test.war}
```

**Comment by joakime** [comment]
In this strange failure case, the logs show ...

> TOUCH of test.war

The existing webapp is starting shut down

> 2024-04-17 16:09:40.043:INFO :org.eclipse.jetty.ee10.log4j2.LogContextListener:Scanner-0-1: contextDestroyed(): jakarta.servlet.ServletContextEvent[source=ServletContext@oeje10w.WebAppContext@a486d78{test,/test,b=file:///home/jenkins/agent/workspace/jetty.project_PR-11637/tests/test-distribution/test-ee10-distribution/target/tests/oejetd.DisableUrlCacheTest/testReloadWebAppWithLog4j2/work/test/webapp/,a=AVAILABLE,h=oeje10s.SessionHandler@cdc3aae{STOPPING}}{/home/jenkins/agent/workspace/jetty.project_PR-11637/tests/test-distribution/test-ee10-distribution/target/tests/oejetd.DisableUrlCacheTest/testReloadWebAppWithLog4j2/webapps/test.war}]

We see the context was destroyed on `WebAppContext@a486d78{test`
Then the same webapp is stopped.

> 2024-04-17 16:09:40.048:INFO :oejes.ServletContextHandler:Scanner-0-1: Stopped oeje10w.WebAppContext@a486d78{test,/test,b=null,a=AVAILABLE,h=oeje10s.SessionHandler@cdc3aae{STOPPED}}{/home/jenkins/agent/workspace/jetty.project_PR-11637/tests/test-distribution/test-ee10-distribution/target/tests/oejetd.DisableUrlCacheTest/testReloadWebAppWithLog4j2/webapps/test.war}

Then the deployer sees the new app and starts from the `addApp` process.

> 2024-04-17 16:09:40.050:INFO :oejd.DeploymentManager:Scanner-0-1: addApp: App@1e33b9e7[ee10,null,/home/jenkins/agent/workspace/jetty.project_PR-11637/tests/test-distribution/test-ee10-distribution/target/tests/oejetd.DisableUrlCacheTest/testReloadWebAppWithLog4j2/webapps/test.xml]

The new webapp is started `WebAppContext@1fcbb594{test,`

> 2024-04-17 16:09:40.322:INFO :oejsh.ContextHandler:Scanner-0-1: Started oeje10w.WebAppContext@1fcbb594{test,/test,b=file:///home/jenkins/agent/workspace/jetty.project_PR-11637/tests/test-distribution/test-ee10-distribution/target/tests/oejetd.DisableUrlCacheTest/testReloadWebAppWithLog4j2/work/test/webapp/,a=AVAILABLE,h=oeje10s.SessionHandler@e664844{STARTED}}{/home/jenkins/agent/workspace/jetty.project_PR-11637/tests/test-distribution/test-ee10-distribution/target/tests/oejetd.DisableUrlCacheTest/testReloadWebAppWithLog4j2/webapps/test.war}

The startup of this new context has a logging output on the LogContextListener contextInitialized()

> 2024-04-17 16:09:40.325:INFO :org.eclipse.jetty.ee10.log4j2.LogContextListener:Scanner-0-1: contextInitialized(): jakarta.servlet.ServletContextEvent[source=ServletContext@oeje10w.WebAppContext@1fcbb594{test,/test,b=file:///home/jenkins/agent/workspace/jetty.project_PR-11637/tests/test-distribution/test-ee10-distribution/target/tests/oejetd.DisableUrlCacheTest/testReloadWebAppWithLog4j2/work/test/webapp/,a=AVAILABLE,h=oeje10s.SessionHandler@e664844{STARTED}}{/home/jenkins/agent/workspace/jetty.project_PR-11637/tests/test-distribution/test-ee10-distribution/target/tests/oejetd.DisableUrlCacheTest/testReloadWebAppWithLog4j2/webapps/test.war}]

The servlet in the webapp is even started, and shows its LogServlet.init() being run

> 2024-04-17 16:09:40.332:INFO :org.eclipse.jetty.ee10.log4j2.LogServlet:Scanner-0-1: #### init()

The webapp is reported as started (again?) `WebAppContext@1fcbb594{test`

> 2024-04-17 16:09:40.332:INFO :oejes.ServletContextHandler:Scanner-0-1: Started oeje10w.WebAppContext@1fcbb594{test,/test,b=file:///home/jenkins/agent/workspace/jetty.project_PR-11637/tests/test-distribution/test-ee10-distribution/target/tests/oejetd.DisableUrlCacheTest/testReloadWebAppWithLog4j2/work/test/webapp/,a=AVAILABLE,h=oeje10s.SessionHandler@e664844{STARTED}}{/home/jenkins/agent/workspace/jetty.project_PR-11637/tests/test-distribution/test-ee10-distribution/target/tests/oejetd.DisableUrlCacheTest/testReloadWebAppWithLog4j2/webapps/test.war}

Then the HTTP request to path `/test/foo` is performed and we get a 404?

**Comment by lorban** [comment]
Two other failure instances:
 - https://jenkins.webtide.net/job/jetty.project/job/jetty-12.0.x/1968/testReport/junit/org.eclipse.jetty.ee10.tests.distribution/DisableUrlCacheTest/Parallel_Stage___Build___Test___JDK22___testReloadWebAppWithLog4j2/
 - https://jenkins.webtide.net/job/jetty.project/job/jetty-12.0.x/1969/testReport/junit/org.eclipse.jetty.ee10.tests.distribution/DisableUrlCacheTest/Parallel_Stage___Build___Test___JDK21___testReloadWebAppWithLog4j2/

**Comment by gregw** [comment]
This was not fixed by #11670, as that is still waiting for an event that happens during starting, not started.    Actual fix is in #11691

**Comment by joakime** [comment]
PR #11691 merged
